packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -575,52 +575,49 @@ function createPatchedFetcher(
+              incrementalCache &&
               cacheKey &&
               (isCacheableRevalidate || requestStore?.serverComponentsHmrCache)
             ) {
-              const bodyBuffer = Buffer.from(await res.arrayBuffer())
-
-              const cachedFetchData = {
-                headers: Object.fromEntries(res.headers.entries()),
-                body: bodyBuffer.toString('base64'),
-                status: res.status,
-                url: res.url,
-              }
-
-              requestStore?.serverComponentsHmrCache?.set(
-                cacheKey,
-                cachedFetchData
-              )
+              res
+                .clone()
+                .arrayBuffer()
+                .then(async (arrayBuffer) => {
+                  const bodyBuffer = Buffer.from(arrayBuffer)
+
+                  const cachedFetchData = {
+                    headers: Object.fromEntries(res.headers.entries()),
+                    body: bodyBuffer.toString('base64'),
+                    status: res.status,
+                    url: res.url,
+                  }
 
-              if (isCacheableRevalidate) {
-                try {
-                  await staticGenerationStore.incrementalCache.set(
+                  requestStore?.serverComponentsHmrCache?.set(
                     cacheKey,
-                    {
-                      kind: 'FETCH',
-                      data: cachedFetchData,
-                      revalidate: normalizedRevalidate,
-                    },
-                    {
-                      fetchCache: true,
-                      revalidate: finalRevalidate,
-                      fetchUrl,
-                      fetchIdx,
-                      tags,
-                    }
+                    cachedFetchData
                   )
-                } catch (err) {
-                  console.warn(`Failed to set fetch cache`, input, err)
-                }
-              }
 
-              const response = new Response(bodyBuffer, {
-                headers: new Headers(res.headers),
-                status: res.status,
-              })
-              Object.defineProperty(response, 'url', { value: res.url })
-              return response
+                  if (isCacheableRevalidate) {
+                    await incrementalCache.set(
+                      cacheKey,
+                      {
+                        kind: 'FETCH',
+                        data: cachedFetchData,
+                        revalidate: normalizedRevalidate,
+                      },
+                      {
+                        fetchCache: true,
+                        revalidate: finalRevalidate,
+                        fetchUrl,
+                        fetchIdx,
+                        tags,
+                      }
+                    )
+                  }
+                })
+                .catch((error) =>
+                  console.warn(`Failed to set fetch cache`, input, error)
```

Comment:
Shouldn't this be an error since that's what it used to be?

Reply:
It was a warning before: https://github.com/vercel/next.js/pull/68447/files#diff-4c54e369ddb9a2db1eed95fe1d678f94c8e82c540204475d42c78e49bf4f223aL614 The only difference is that we now also catch the HMR cache set, which we should have done from the beginning.

Reply:
Awaiting the body buffer was not caught before. But I would argue that it should be, since this is only done for the purpose of setting the cache.